kafka/src/main/java/org/axonframework/kafka/eventhandling/consumer/AsyncFetcher.java

```diff
@@ -116,119 +121,176 @@ private AsyncFetcher(Builder<K, V> builder) {
+     *            KafkaMessageConverter}
      */
     public static final class Builder<K, V> {
 
-        private final ConsumerFactory<K, V> consumerFactory;
+        private ConsumerFactory<K, V> consumerFactory;
         private Supplier<Buffer<KafkaEventMessage>> bufferFactory = SortedKafkaMessageBuffer::new;
-        private KafkaMessageConverter<K, V> converter =
-                (KafkaMessageConverter<K, V>) new DefaultKafkaMessageConverter(new XStreamSerializer());
+        private ExecutorService executorService =
+                Executors.newCachedThreadPool(new AxonThreadFactory("AsyncFetcher-pool-thread"));
+        private boolean requirePoolShutdown = true;
+        @SuppressWarnings("unchecked")
+        private KafkaMessageConverter<K, V> messageConverter =
+                (KafkaMessageConverter<K, V>) DefaultKafkaMessageConverter.builder()
+                                                                          .serializer(new XStreamSerializer())
+                                                                          .build();
         private String topic = "Axon.Events";
+        private BiFunction<ConsumerRecord<K, V>, KafkaTrackingToken, Void> consumerRecordCallback = (r, t) -> null;
         private long pollTimeout = 5_000;
 
-        private ExecutorService pool = newCachedThreadPool(new AxonThreadFactory("AsyncFetcher-pool-thread"));
-        private BiFunction<ConsumerRecord<K, V>, KafkaTrackingToken, Void> callback = (r, t) -> null;
-        private boolean requirePoolShutdown = true;
 
-        private Builder(ConsumerFactory<K, V> consumerFactory) {
-            Assert.notNull(consumerFactory, () -> "ConsumerFactory may not be null");
+        /**
+         * Sets the {@link ConsumerFactory} to be used by this {@link Fetcher} implementation to create {@link Consumer}
+         * instances.
+         *
+         * @param consumerFactory a {@link ConsumerFactory} to be used by this {@link Fetcher} implementation to create
+         *                        {@link Consumer} instances
+         * @return the current Builder instance, for a fluent interfacing
+         */
+        public Builder<K, V> consumerFactory(ConsumerFactory<K, V> consumerFactory) {
+            assertNonNull(consumerFactory, "ConsumerFactory may not be null");
             this.consumerFactory = consumerFactory;
+            return this;
+        }
+
+        /**
+         * Instantiate a {@link DefaultConsumerFactory} with the provided {@code consumerConfiguration}. Used by this
+         * {@link Fetcher} implementation to create {@link Consumer} instances.
+         *
+         * @param consumerConfiguration a {@link DefaultConsumerFactory} with the given {@code consumerConfiguration},
+         *                              to be used by this {@link Fetcher} implementation to create {@link Consumer}
+         *                              instances
+         * @return the current Builder instance, for a fluent interfacing
```

Comment:
Same as earlier (https://github.com/AxonFramework/AxonFramework/pull/768#discussion_r217031691)

`for a fluent interfacing` or `for fluent interfacing`

Reply:
Argh darn, that one came through in my template again....

Reply:
Aah, luckily we have "find & replace"